### DISCUSSION TITLE
Exporting Android Release with Flash Builder 4.7

### DISCUSSION THREAD

**Discussion by aliwithay** [airsdk/Adobe-Runtime-Support#1024] [Q&A]
We are trying to update an older android app built with flash builder but there are two major issues we are having, getting the android release to export. 

1. Trying to export a 64-bit apk: Since google requires having both 32 and 64 bit apk files for apps on the play store, we are trying to export a 64-bit version of the apk. Exporting a 32-bit version seems to work just fine, however, even after updating the adt.cfg to force armv8 architecture, as well as adding the "-arch armv8" flag to the build command (both suggested by the AIR Release Notes), the exported apk shows as armv7a. Is there anything else we can try to get the build to work correctly?

2. Creating an Android App Bundle: We also considered just using the App Bundle so we dont have to export two separate apk files, and also to future proof the app. However, when trying to export a build with the adt.cfg updated to enable aab support, and adding the AndroidSdk and Java paths, we seem to be running into issues with gradle, as seen below. Any explanation on what might be causing this issue or a decent guide on getting aab files exported from Flash Builder would be very much appreciated!
 
![image](https://user-images.githubusercontent.com/27898174/127575057-631a1086-0e13-430c-85bb-0fafdc58422a.png)
![image](https://user-images.githubusercontent.com/27898174/127575096-add3a8b4-9454-4731-bfdb-6d2745af14ec.png)

**Comment by ajwfrost** [comment]
Hi

For generating an APK that has 64-bit ARM, from Flash Builder, then I'm not sure why it doesn't work as it looks like you're doing everything right. Just to confirm that you're using both of the settings within the config file - there's an 'override' one and a 'default' one and I believe for Flash Builder you'd need to use the 'override' one, but it's probably safest to set both of them. Perhaps just double-check that it's actually calling the right SDK path when doing the export?

For that bundle error `unable to make field private` - it may be a Java compatibility issue with Gradle. What version of Java are you running? it may be worth using a slightly older version. Alternatively, for these sorts of things, if you set your lib\adt.cfg file so that it has `KeepAndroidStudioOutput=true` you can then open the resulting AndroidStudioProject folder directly within Android Studio, and from there you can potentially update the Gradle version which is another possible solution.

thanks

**Reply by aliwithay** [reply]
Hey! 

Both "override" and "default" flags are set in the adt.cfg file. And I did check the SDK path. and it Flash Builder is using the correct SDK. I will try to do some more research and see why the output is not armv8.

For the bundle, I did try to import the project files into android studio, but trying to build the project in android studio seems to still fail with the error in the picture attached below. I tried to do some research on what the error means, but was unable to find a solution to this!

![image](https://user-images.githubusercontent.com/27898174/127933956-1dc8643b-ffdb-4509-ad0c-87a1712722c1.png)

**Reply by ajwfrost** [reply]
I saw that message previously, when there was an AIR native extension that had an "AndroidManifest.xml" file included in it. I don't think those files should be included in the ANEs: if there are any settings necessary for the manifest file then these can be added via the main Application descriptor file.

But if this is a common situation then it may be that we can look to detect these files and merge in the contents - @marchbold are you able to comment on this please, are there situations where an ANE would have an AndroidManifest.xml file that we need to take some settings from?

**Reply by aliwithay** [reply]
It is definitely the ANEs with AndroidManifest.xml files that seem to be causing this issue. Flash Builder uses AGP 4.1.1. Upgrading the AGP to 7.0.0 (as recommended by Android Studio) gave me a more detailed error message, pointing to the two AndroidManifest.xml files in ANEs. 

I will try removing the files and report back. 

EDIT: Removing the AndroidManifest.xml files from the ANEs seems to allow me to compile and export builds with android studio.

**Reply by marchbold** [reply]
@ajwfrost As far as I am aware there should be no reason to have an AndroidManifest.xml file in an ANE. 

I know some people were discussing including the android manifest from an AAR in an ANE to use as a manifest merge but we always advised against this as we predicted a situation like this! :)

**Reply by ajwfrost** [reply]
Thank!

**Reply by aliwithay** [reply]
Thanks @marchbold!

So it looks like removing the AndroidManifest.xml from the ANEs seems to have done the trick and I am able to build and export an AAB from Android Studio. Having successfully gotten that build working, I went back to working with flash builder with the new ANEs, however, it does not move past the progress status seen in the screenshot. The same issue occurs when trying to debug on device, as seen in the second screenshot.

This issue only happens when trying to export an AAB. Exporting and debugging, with the aab options disabled in adt.cfg, seems to work just fine!

![image](https://user-images.githubusercontent.com/27898174/128746333-1c548ed2-5ad7-4a82-8a95-60a26c660723.png)
![image](https://user-images.githubusercontent.com/27898174/128746791-bc043287-0fc3-4c1a-ac09-85d1ff6a6578.png)
